sys/net/gnrc/link_layer/lorawan/gnrc_lorawan.c

```diff
@@ -69,6 +69,11 @@ void gnrc_lorawan_set_rx2_dr(gnrc_lorawan_t *mac, uint8_t rx2_dr)
                         GNRC_LORAWAN_DL_RX2_DR_MASK;
 }
 
+void gnrc_lorawan_set_adr(gnrc_lorawan_t *mac, bool adr)
+{
+    mac->mlme.adr = adr;
+}
+
```

Comment:
Please use the `gnrc_lorawan_mlme_request` function (passing MIB_ADR) to enable/disable the ADR

Reply:
ah, I see it's integrated below.
Then, I would suggest to simply drop this function, since it's only used in one point and doesn't require any special logic.

Reply:
Alternatively, this function could still be here, but it should reset the counters when the feature is turned on

Reply:
Yup. Updated.